tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -124,7 +131,8 @@ public void InjectPathwayContext<TCarrier>(PathwayContext? context, TCarrier hea
                 previousContext = PreviousCheckpoint.Value.Context;
             }
 
-            var edgeStartNs = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
+            var nowNs = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
+            var edgeStartNs = previousContext == null && timeInQueueMs > 0 ? nowNs - (timeInQueueMs * 1000_000) : nowNs;
```

Comment:
tiny formatting nit! 😄 
```suggestion
            var edgeStartNs = previousContext == null && timeInQueueMs > 0 ? nowNs - (timeInQueueMs * 1_000_000) : nowNs;
```